storage/storage_dest.go

```diff
@@ -307,6 +307,9 @@ func (s *storageImageDestination) PutBlobPartial(ctx context.Context, chunkAcces
 // If the blob has been successfully reused, returns (true, info, nil).
 // If the transport can not reuse the requested blob, TryReusingBlob returns (false, {}, nil); it returns a non-nil error only on an unexpected failure.
 func (s *storageImageDestination) TryReusingBlobWithOptions(ctx context.Context, blobinfo types.BlobInfo, options private.TryReusingBlobOptions) (bool, private.ReusedBlob, error) {
+	if options.RequiredCompression != nil && (options.RequiredCompression != blobinfo.CompressionAlgorithm) {
```

Comment:
(It’s an “interesting” question whether a storage destination, which pretends to store compressed blobs but always actually stores uncompressed variants, should care about `RequiredCompression`. In practice it makes no difference at all, I think.)